Test/DurableTask.AzureStorage.Tests/TestTablePartitionManager.cs

```diff
@@ -0,0 +1,723 @@
+                    WorkerId = i.ToString(),
+                    UseTablePartitionManagement = true,
+                };
+                services[i] = new AzureStorageOrchestrationService(settings);
+                taskHubWorkers[i] = new TaskHubWorker(services[i]);
+                taskHubWorkers[i].AddTaskOrchestrations(typeof(LongRunningOrchestrator));
+            }
+
+            // Create 50 orchestration instances.
+            var client = new TaskHubClient(services[0]);
+            var createInstanceTasks = new Task<OrchestrationInstance>[InstanceCount];
+            for (int i = 0; i < InstanceCount; i++)
+            {
+                createInstanceTasks[i] = client.CreateOrchestrationInstanceAsync(typeof(LongRunningOrchestrator), input: null);
+            }
+            OrchestrationInstance[] instances = await Task.WhenAll(createInstanceTasks);
+
+            await taskHubWorkers[0].StartAsync();
+            //Ensure worker 0 acquired the partition.
+            WaitForCondition(
+                timeout: TimeSpan.FromSeconds(2),
+                condition: () => services[0].ListTableLeases().Single().CurrentOwner == "0");
+            await taskHubWorkers[1].StartAsync();
+
+            using var cts = new CancellationTokenSource();
+            {
+                //Kill worker[0] and start a new worker. 
+                //The new worker will take over the partitions of worker[0].
+                services[0].KillPartitionManagerLoop();
+                WaitForCondition(
+                    timeout: TimeSpan.FromSeconds(40),
+                    condition: () => services[0].ListTableLeases().Single().CurrentOwner == "1");
+
+                //After worker[1] takes over the lease, restart the worker[0].
+                services[0].SimulateUnhealthyWorker(cts.Token);
+
+                //Wait one second for worker0 to remove the lost partitions' control queue.
+                await Task.Delay(1000);
+                Assert.AreEqual(1, services[1].OwnedControlQueues.Count());
+                Assert.AreEqual(0, services[0].OwnedControlQueues.Count());
+
+                //Check all the instances could be processed successfully. 
+                OrchestrationState[] states = await Task.WhenAll(
+                    instances.Select(i => client.WaitForOrchestrationAsync(i, TimeSpan.FromSeconds(30))));
+                Assert.IsTrue(
+                    Array.TrueForAll(states, s => s?.OrchestrationStatus == OrchestrationStatus.Completed),
+                    "Not all orchestrations completed successfully!");
+            }
+            await taskHubWorkers[1].StopAsync();
+        }
+
+        /// <summary>
+        /// End to End test with four workers and four partitions.
+        /// Ensure that no worker listens to the same queue at the same time during the balancing process.
+        /// Ensure that all instances could be process sucessfully.
+        /// </summary>
+        /// <returns></returns>
```

Comment:
nit: some phrasing suggestions
```suggestion
        /// <summary>
        /// End to End test with four workers and four partitions.
        /// Ensure that no worker listens to the same queue at the same time during the balancing process.
        /// Ensure that all instances could be processsed successfully.
        /// </summary>
        /// <returns></returns>
```